development/extensions/tutorial_notifications.rst

```diff
@@ -0,0 +1,955 @@
+    */
+   public function find_users_for_notification($data, $options = [])
+   {
+       return $this->check_user_notification_options([$data['user_id']], $options);
+   }
+
+.. note::
+
+    The ``forum_id`` variable below is not set in this Tutorial, just shown as an example.
+
+.. code-block:: php
+
+   public function_find_users_for_notification($data, $options = [])
+   {
+       return $this->get_authorised_recipients([$data['user_id']], $data['forum_id'], $options);
+   }
+
+users_to_query
+--------------
+This function should return an array of user identifiers.
+Identifiers of users that need to be queried for this notification to be displayed to the user that is notified.
+For example users whose username need to be displayed, or the user whose avatar will be shown next to the notification text.
+
+.. code-block:: php
+
+   /**
+    * Users needed to query before this notification can be displayed.
+    *
+    * @return array				Array of user identifiers to query.
+    */
+   public function users_to_query()
+   {
+       return [$this->get_data('sender_id')];
+   }
+
+get_avatar
+----------
+The user identifiers returned by the users_to_query_ function are added to the User loader object.
+Unfortunately, that object is not available by default in the `Base services`_.
+However, we have added that in the `Additional services`_ section, so we can use it anyway!
+
+So we use the convenient function, applicably named, ``get_avatar()`` that is available in the User loader.
+And we supply three parameters. The first being the user identifier from whom we want to show the avatar.
+The second is a boolean, indicating that the user does not have to be queried from the database, as that is already done.
+And the third is also a boolean, indicating that the avatar image should be lazy loaded in the HTML.
+Seeing it is in a dropdown and not visible immediately, lazy loading is more beneficial for a page's load time.
+
+If you do not wish to display an avatar next to the notification text, you can omit this function all together.
+
+.. code-block:: php
+
+   /**
+    * Get the user's avatar.
+    *
+    * @return string				The HTML formatted avatar
+    */
+   public function get_avatar()
+   {
+       return $this->user_loader->get_avatar($this->get_data('sender_id'), false, true);
+   }
+
+get_title
+---------
+This should return the main text for the notification.
+Usually the action that was taken why this user received a notification.
+Commonly the action is made bold and everything else is regular, as shown in the `language file`_.
+
+.. code-block:: php
+
+   /**
+    * Get the title of this notification.
+    *
+    * @return string				The notification's title
+    */
+   public function get_title()
+   {
+       return $this->language->lang(
+           'VENDOR_EXTENSION_NOTIFICATION_SAMPLE_TITLE',
+           $this->user_loader->get_username($this->get_data('sender_id'), 'no_profile')
```

Comment:
Add a note that in this case, the lang string has 2 variables that are inserted into its placeholders